src/python/pants/option/options.py

```diff
@@ -141,8 +141,18 @@ def create(
         # We need registrars for all the intermediate scopes, so inherited option values
         # can propagate through them.
         complete_known_scope_infos = cls.complete_scopes(known_scope_infos)
+
+        config_to_pass = None if native_options_config_discovery else config.sources()
```

Comment:
This is just moved up from below, so we can create the native_parser earlier and use it to get the args to split.